Arriba/Arriba.Web/parts/Index.jsx

```diff
@@ -1,22 +1,78 @@
 import "./Index.scss";
 
+import EventedComponent from "./EventedComponent";
+import Mru from "./Mru";
 import ErrorPage from "./ErrorPage";
+import SearchHeader from "./SearchHeader";
+import Tabs from "./Tabs";
+import SearchBox from "./SearchBox";
 import Search from "./Search";
 import Grid from "./Grid";
 window.configuration = require("../configuration/Configuration.jsx").default;
 
-class Index extends React.Component {
+// For schema detection and possible migration.
+localStorage.setItem("version", 1);
+
+class Index extends EventedComponent {
     constructor(props) {
         super(props);
+        this.mru = new Mru();
+
+        this.params = getQueryStringParameters();
+        const table = this.params.t;
+        const columns = getParameterArrayForPrefix(this.params, "c");
+
+        if (table) {
+            localStorage.mergeJson("table-" + table, ({
+                columns: columns.emptyToUndefined(),
+                sortColumn: this.params.ob || undefined, // Filter out empty strings.
+                sortOrder: this.params.so || undefined
+            }).cleaned);
+        }
+        const query = this.params.q || "";
+
         this.state = {
-            allBasics: [],
-            blockingErrorStatus: null
+            blockingErrorStatus: null,
+            allBasics: {},
+            query: query,
+            debouncedQuery: query, // Required to trigger getCounts.
+            currentTable: table,
+            userSelectedTable: table,
+            mode: window.location.pathname.startsWith("/Grid.html") // false means Search, true means Grid.
```

Comment:
It seems like we should make mode more than a boolean expecting additional modes later (timeline/graph?)

Reply:
I didn't seem hard to implement that later, so I went with the most basic solution now.